apps/meteor/client/views/room/contextualBar/RoomMembers/AddUsers/ValidateMatrixInvitedUsers/useMatrixIdValidation.tsx

```diff
@@ -0,0 +1,48 @@
+import type { IRoom } from '@rocket.chat/core-typings';
+import { useMutableCallback } from '@rocket.chat/fuselage-hooks';
+import { useSetModal, useMethod, useToastMessageDispatch, useTranslation } from '@rocket.chat/ui-contexts';
+import React, { useMemo } from 'react';
+
+import ValidateMatrixIdModal from './ValidateMatrixIdModal';
+
+export type useMatrixIdValidationProps = {
+	rid: IRoom['_id'];
+	matrixIdVerifiedStatus: Map<string, string>;
+	saveAction: () => void;
+	onClickBack: () => void;
+	reload: () => void;
+	users: string[];
+};
+
+export const useMatrixIdValidation = ({
```

Comment:
If the previous comments are respected, I suggest this gets renamed to `useAddMatrixUsers`